reach-whitepaper.md

```diff
@@ -0,0 +1,1019 @@
+Specifically, these datasets exhibit a power-law distribution, albeit with
+differing shape parameters. 
+In our observations, it was noted that capping on the client side does not
+affect the shape of the power-law distribution, indicating that campaign
+impressions are dropped by capping with equal probability.
+In contrast, capping does have a significant impact on the volume reports from
+clients.
+The rate of reports dropped by capping, i.e., the capping discount rate, follows
+a power-law distribution with respect to the capping value.
+Figure 2 displays a representative discount rate versus capping plot and
+corresponding curve fitting.
+
+Furthermore, we examined the impact of the time window on the reach. Reach does
+not demonstrate a linear relationship with time window size.
+Instead, it exhibits a diminishing return compared to linear growth as
+illustrated in Figure 3. 
+This discount follows a power-law function relative to the size of the time
+window.
+
+![Reach estimates receive a discount as a function of cap value.](figs/cap-discount.png)
+
+![Reach does not increase linearly with time windows size.](figs/window-discount.png)
+
+
+```python
+def sample_slices(
+   number_of_days,
+   cap,
+   dataset_power_law_shape,
+   n_samples
+) -> list[SliceSize]:
+"""Generates a synthetic dataset in slices (tuples of capped and uncapped numbers).
+
+
+ Args:
+   number_of_days: what time window the query is using.
+   cap: the maximal number of slices a user might contribute.
+   dataset_power_law_shape: dataset specific power-law shape param.
+   n_samples: number of slices to generate. 
+ """
+
+
+  # sample reach distribution from power-law distribution. 
+  reach_1_day = sample_discrete_power_law(
+    b=dataset_power_law_shape, 
+    n_samples=n_samples,
+    ...)
+  # approximate impact of capping.
+  # compute probability of being reported (success, not impacted by capped) 
+  # after capping.
+  p_reported = (1 - cap_discount_rate(cap=cap))
+  # for a slice of true size n, each contribution has p_reported probability to
+  # be reported. Thus after capping, observed contributions become 
+  # np.random.binomial(n, p) (there may be faster approximations) 
+  reach_1_day_capped =  np.random.binomial(reach_1_day, p_reported)
+
+
+  # approximate impact of time window
+  reach_n_days = reach_1_day * number_of_days * window_discount_rate(number_of_days)
+  reach_n_days_capped = reach_1_day_capped * number_of_days * window_discount_rate(number_of_days)
+
+
+  # round to nearest int 
+  reach_n_days = np.round(reach_n_days).astype(int)
+  reach_n_days_capped = np.round(reach_n_days_capped).astype(int)
+
+
+  return [
+    (uncapped, capped) 
+    for uncapped, capped in zip(reach_n_days, reach_n_days_capped)
+  ]
+```
+
+
+Listing 1: pseudocode to generate synthetic data. 
+
+
+<table>
+  <tr>
+   <td style="background-color: #cccccc">
+    Level
+<p>
+
+    (granularity)
```

Comment:
This appears to have been treated as a code block

Reply:
right. replacing <p> with <br> solved it.